Automattic/jetpack
# Google Analytics: inform users of the feature removal

## Proposed changes:
* The feature has been removed from Jetpack, we update the GA deprecation notices to reflect that.
* Make all the notices dismissible.

### Other information:


## Jetpack product discussion
https://github.com/Automattic/vulcan/issues/427

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

### Prepare the site
First, we need to get into the situation of the GA module being active, but removed.
1. Setup a JN site with Beta set to the "Latest Stable" (13.6).
2. Connection Jetpack, go to "Settings -> Traffic -> Google Analytics", follow the "Upgrade" link to purchase the required plan.
3. Go back to "Settings -> Traffic -> Google Analytics", follow the link to Calypso, and activate GA with a random Measurement ID (e.g. "G-12345").
4. Go to Jetpack Beta and checkout this PR's branch (`update/ga-deprecation-notices`).
5. Now there's no GA module, but it should still be stored as "active". Please check if that's the case by running `wp option get jetpack_active_modules` via SSH.

### Check the notices
The "GA has been removed" notices should appear in 3 places:
- Non-Jetpack WP-admin pages (Dashboard, Plugins, etc.)
- Jetpack Dashboard and Settings
- My Jetpack

To test them out, please do the following:
1. Follow the link to the docs, confirm it leads to the GA support page.
2. Click "X" to dismiss the notice on each notice individually to confirm it disappears in all 3 places. Then bring it back (remove the cookie `jetpack_deprecate_dismissed[jetpack-ga-admin-removal-notice]`) and test it on another notice.
